### PR TITLE
re-factored pagination to work with new back-end, changed range to 5 …

### PR DESCRIPTION
**Pull request by charliew-rsi**
…articles per page

### CODE REVIEW

File: src/components/ArticleList.js [hunk 1]
```diff
@@ -1,36 +1,34 @@
...
   if (props.articles.length === 0) {
-    return (
-      <div className="article-preview">
-        No articles are here... yet.
-      </div>
-    );
+    return <div className="article-preview">No articles are here... yet.</div>;
   }
 
+  console.log(props.currentPage);
```
Comment (wesdollar) [inline src/components/ArticleList.js]: Remove `console.log`

File: src/components/ArticleList.js [hunk 2]
```diff
@@ -1,36 +1,34 @@
...
+
+  const pageArticles = props.articles.slice(
```
Comment (wesdollar) [inline src/components/ArticleList.js]: Destructure `articles` and fix all references

File: src/components/ArticleList.js [hunk 3]
```diff
@@ -1,36 +1,34 @@
...
+    props.currentPage,
+    props.currentPage + 5
```
Comment (wesdollar) [inline src/components/ArticleList.js]: destructure

File: src/components/ArticleList.js [hunk 4]
```diff
@@ -1,36 +1,34 @@
...
+      {pageArticles.map(article => {
+        return <ArticlePreview article={article} key={article.slug} />;
+      })}
 
       <ListPagination
         pager={props.pager}
-        articlesCount={props.articlesCount}
-        currentPage={props.currentPage} />
+        articlesCount={props.articles.length}
+        currentPage={props.currentPage}
```
Comment (wesdollar) [inline src/components/ArticleList.js]: destructure

File: src/components/ArticleList.js (hunk 1 above)
Comment (charliew-rsi) [reply src/components/ArticleList.js]: resolved

File: src/components/ArticleList.js (hunk 2 above)
Comment (charliew-rsi) [reply src/components/ArticleList.js]: resolved

File: src/components/ArticleList.js (hunk 3 above)
Comment (charliew-rsi) [reply src/components/ArticleList.js]: resolved

File: src/components/ArticleList.js (hunk 4 above)
Comment (charliew-rsi) [reply src/components/ArticleList.js]: resolved

**Pull request merged**